### PR TITLE
[KOGITO-8298] Retrieving Kaniko task only if it is present into the P…

### PR DESCRIPTION
**Pull request by davidesalerno**
…latform CR

See: [KOGITO-8298](https://issues.redhat.com/browse/KOGITO-8298)

Signed-off-by: Davide Salerno <dsalerno@redhat.com>

**Description of the change:**

**Motivation for the change:**

**Checklist**

- [ ] Add or Modify a unit test for your change
- [ ] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: builder/builder.go [hunk 1]
```diff
@@ -88,11 +88,23 @@ func (b *Builder) ScheduleNewBuildWithContainerFile(workflowName string, imageTa
 }
 
 func (b *Builder) ScheduleNewKanikoBuildWithContainerFile(workflowName string, imageTag string, workflowDefinition []byte, build api.BuildSpec) (*api.Build, error) {
-	ib := b.getImageBuilderForKaniko(workflowName, imageTag, workflowDefinition, build.Tasks[0].Kaniko)
+	task := findKanikoTask(build.Tasks)
```
Comment (spolti) [inline builder/builder.go:91]: I think it is better to return a bool flag on findKanikoTask as well, this way you can avoid proceeding with the code if the task does not exists.

Comment (ricardozanini) [reply builder/builder.go:91]: That makes sense. We could even call the method within the conditional.

Comment (davidesalerno) [reply builder/builder.go]: @spolti I followed you suggestion and added an additional bool return value that I use to skip the code execution if no Kaniko task is found.

File: builder/builder.go [hunk 2]
```diff
@@ -88,11 +89,26 @@ func (b *Builder) ScheduleNewBuildWithContainerFile(workflowName string, imageTa
 }
 
 func (b *Builder) ScheduleNewKanikoBuildWithContainerFile(workflowName string, imageTag string, workflowDefinition []byte, build api.BuildSpec) (*api.Build, error) {
-	ib := b.getImageBuilderForKaniko(workflowName, imageTag, workflowDefinition, build.Tasks[0].Kaniko)
+	task, found := findKanikoTask(build.Tasks)
```
Comment (spolti) [inline builder/builder.go:92]: this:
```
task, found := findKanikoTask(build.Tasks)
if !found {
	return nil, fmt.Errorf("Unable to find KanikoTask in Build CR for workflow %s", workflowName)
}
```

can be turned in something like this: 
```
if task, found := findKanikoTask(build.Tasks); !found {
	return nil, fmt.Errorf("Unable to find KanikoTask in Build CR for workflow %s", workflowName)
} else {
	ib := b.getImageBuilderForKaniko(workflowName, imageTag, workflowDefinition, task)
	ib.WithTimeout(5 * time.Minute)
	return b.BuildImage(ib.Build())
}
```

Comment (spolti) [reply builder/builder.go:92]: it does not change nothing, just sabe a few lines :)

Comment (davidesalerno) [reply builder/builder.go]: Saved one line of code ;-)

**Review by desmax74 — APPROVED** [review]

**Review by spolti — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

**Comment by spolti** [comment]
@radtriste do you want to take a look before we merge it?

**Review by radtriste — APPROVED** [review]

**Pull request merged**